onedrive: 2.4.25 -> 2.5.0-rc3

AndersonTorres opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

AndersonTorres committed:
onedrive: remove srgom from meta.maintainers

srgom is inactive

AndersonTorres committed:
onedrive: migrate to by-name

AndersonTorres committed:
onedrive: rework

- finalAttrs
- lib.enable* flags
- ghet rid of pname
- test version
- get rid of nested with
- nixfmt-rfc-style

AndersonTorres committed:
onedrive: 2.4.25 -> 2.5.0-rc3

AndersonTorres committed:
onedrive: add AndersonTorres as maintainer

bertof reviewed (approved):
Seems fine to me. Thank you for the PR.

AndersonTorres commented:
Is there a bug in the tester??

> Version string '2.5.0-rc3' not found! The output was: onedrive v2.5.0-rc3 error: builder for '/nix/store/9vz57lrzgzf75lpc8sd1pfjbx5ps2zfl-onedrive-2.5.0-rc3-test-version.drv' failed with exit code 1; last 3 log lines: 

> Version string '2.5.0-rc3' not found! > The output was: 
> onedrive v2.5.0-rc3
> For full logs, run 'nix-store -l /nix/store/9vz57lrzgzf75lpc8sd1pfjbx5ps2zfl-onedrive-2.5.0-rc3-test-version.drv'. 

View more details on OfBorg

AndersonTorres committed:
onedrive: fixup version test

Since the test uses `grep --word-regexp --fixed-string`, we should use the full
version style.

bertof reviewed (approved):
Seems good to me
